app/src/main/java/com/nexters/mytine/ui/home/HomeViewModel.kt

```diff
@@ -75,4 +103,49 @@ internal class HomeViewModel @ViewModelInject constructor(
     private fun iconItems(): List<IconItem> {
         return listOf("a", "b", "c", "d", "e", "f", "g").map { IconItem(it) }
     }
+
+    private fun getRetrospect() = runBlocking {
+        setDay()
+
+        content.value = ""
+        storedContent = ""
+
+        retrospectRepository.getRetrospect(date).firstOrNull()?.let {
+            content.value = it.contents
+            storedContent = it.contents
+        }
+    }
+
+    fun onClickWriteRetrospect() {
+
+        if (content.value.isNullOrBlank()) {
+            toast.value = resourcesProvider.getString(R.string.write_empty_toast_message)
+            return
+        }
+
+        if (content.value.toString() == storedContent) {
+            toast.value = resourcesProvider.getString(R.string.not_change_toast_message)
+            return
+        }
+
+        viewModelScope.launch {
+            retrospectRepository.updateRetrospect(Retrospect(date, content.value!!))
+            storedContent = content.value!!
+        }
+    }
+
+    private fun checkDataSaved(): Boolean {
+
+        if (content.value != storedContent) {
+            toast.value = "변경된 내용이 있습니다. 회고 저장 후 이동 해 주세요. 다이얼로그로 바꾸기ㅣ!!!"
```

Comment:
하드코딩 strings.xml에 넣어서 resourcesProvider.getString(id)로 써주세용~
```suggestion
            toast.value = resourcesProvider.getString(R.string.foo)
```